mesonbuild/utils/universal.py

```diff
@@ -2361,20 +2355,10 @@ def pickle_load(filename: str, object_name: str, object_type: T.Type[_PL]) -> _P
             f'meson setup {build_dir} --wipe')
     if not isinstance(obj, object_type):
         raise MesonException(load_fail_msg)
-
-    # Because these Protocols are not available at runtime (and cannot be made
-    # available at runtime until we drop support for Python < 3.8), we have to
-    # do a bit of hackery so that mypy understands what's going on here
-    version: str
-    if hasattr(obj, 'version'):
-        version = T.cast('_VerPickleLoadable', obj).version
-    else:
-        version = T.cast('_EnvPickleLoadable', obj).environment.coredata.version
```

Comment:
We cannot remove this, because pickle_load *must* successfully load a pickle created by any version of meson long enough to report a useful error message without crashing.

This was previously discussed when the code was originally introduced: https://github.com/mesonbuild/meson/pull/11063#discussion_r1025709689

I commented at the time:
> We could add a version attribute to build.Build but that doesn't help to detect the problem of people loading build directories created by versions of Meson before 0.64.1